AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/vho-work-hours-table/vho-work-hours-table.component.ts

```diff
@@ -1,20 +1,71 @@
-import { Component, Input, OnInit } from '@angular/core';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
 import { VhoWorkHoursResponse } from '../../../services/clients/api-client';
 
 @Component({
     selector: 'app-vho-work-hours-table',
     templateUrl: './vho-work-hours-table.component.html'
 })
 export class VhoWorkHoursTableComponent implements OnInit {
-    workHours: VhoWorkHoursResponse[];
+    workHours: VhoWorkHoursResponse[] = [];
+    workHoursEndTimeBeforeStartTimeErrors: number[] = [];
+    originalWorkHours: VhoWorkHoursResponse[] = [];
+    isEditing = false;
 
     @Input() set result(value: VhoWorkHoursResponse[]) {
         if (value) {
             this.workHours = value;
         }
     }
 
+    @Output() saveWorkHours: EventEmitter<VhoWorkHoursResponse[]> = new EventEmitter();
+
     ngOnInit(): void {
         console.log('Needs something for sonarcloud. Delete this later');
     }
+
+    cancelEditingWorkingHours() {
+        this.isEditing = false;
+
+        this.workHours = this.originalWorkHours;
+    }
+
+    saveWorkingHours() {
+        this.workHours.forEach((workHour, index) => {
+            if (!workHour.start_time || !workHour.end_time) {
+                return;
+            }
+
+            let workHourArray = workHour.start_time.split(':');
+
+            const startDate = new Date();
+            startDate.setHours(parseInt(workHourArray[0], 10));
+            startDate.setMinutes(parseInt(workHourArray[1], 10));
+
+            workHourArray = workHour.end_time.split(':');
+
+            const endDate = new Date();
+            endDate.setHours(parseInt(workHourArray[0], 10));
+            endDate.setMinutes(parseInt(workHourArray[1], 10));
+
+            if (endDate < startDate) {
+                this.workHoursEndTimeBeforeStartTimeErrors.push(index);
```

Comment:
just wondering - what happens if the endDate is the same as the startDate? I would imagine that shouldn't be allowed either